src/components/dropdown/dropdown.component.ts

```diff
@@ -175,19 +176,26 @@ export default class SlDropdown extends ShoelaceElement {
         return;
       }
 
+      const computeClosestContaining = (element: Element | null | undefined, tagName: string) => {
+        const closest = element?.closest(tagName);
+        if (closest !== null) return closest;
+        return computeClosestContaining((element?.getRootNode() as ShadowRoot).host, tagName);
+      };
```

Comment:
Sorry I didn't catch this the first time around. The §computeClosestContaining()` seems to risk infinite recursion and might break if `getRootNode()` isn't a shadow root. How does this look?

```suggestion
      const computeClosestContaining = (element: Element | null | undefined, tagName: string): Element | null => {
        if (!element) return null;
        
        const closest = element.closest(tagName);
        if (closest) return closest;
        
        const rootNode = element.getRootNode();
        if (rootNode instanceof ShadowRoot) {
          return computeClosestContaining(rootNode.host, tagName);
        }
        
        return null;
      };
```

Reply:
good catch! LGTM, just pushed.